app/services/shop_products_service.rb

```diff
@@ -0,0 +1,76 @@
+# Returns a (paginatable) AR object for the products or variants in stock for a given shop and OC.
+# The stock-checking includes on_demand and stock level overrides from variant_overrides.
+
+class ShopProductsService
```

Comment:
I dont think this name is appropriate. Shop is not a clear concept in OFN, if we use this concept we need to define it first.
I'd prefer for example to keep the previous name OrderCycleDistributedProducts, it makes it very clear what product list this is.
I dont think we can be casual about naming otherwise we end up with things like what we have already in the DB with supplier_id, enterprise_id, distributor_id, hub_id, producer_id and... shop_id (it's a column in the subscription table).

Reply:
or we can call it EnterpriseProductsService

Reply:
OK, I think OrderCycleDistributedProducts is better than EnterpriseProducts.

Reply:
Changed to `OrderCycleDistributedProducts` :+1: